### PR TITLE
Contract creation

### PR DESCRIPTION
**Pull request by kolya182**

### CODE REVIEW

File: src/common/CommonInterfaces.ts [hunk 1]
```diff
@@ -58,7 +58,7 @@ export interface ITransaction {
         blockHash: string | null,
         blockNumber: number,
         from: string,
-        gas: string,
+        gas: number,
```
Comment (vikmeup) [inline src/common/CommonInterfaces.ts]: Does this changes return value?

File: src/common/TransactionParser.ts [hunk 2]
```diff
@@ -68,9 +67,10 @@ export class TransactionParser {
         return newTransaction;
     }
 
-    private extractTransactionData(block: any, transaction: any) {
+    extractTransactionData(block: IBlock, transaction: ITransaction) {
         const from = String(transaction.from).toLowerCase();
-        const to = String(transaction.to).toLowerCase();
+        const to: string = transaction.to === null ? "" : String(transaction.to).toLowerCase();
+        const addresses: string[] = to ? [from, to] : [from];
```
Comment (vikmeup) [inline src/common/TransactionParser.ts]: if `to` == null, then `contract` should exist and you need to save `contract` address

Comment (vikmeup) [inline src/common/TransactionParser.ts]: can you do flatMap instead? `[from, to].flatMap()` - it will remove `null`

Comment (kolya182) [reply src/common/TransactionParser.ts]: Not at this point, later in the code.

File: src/common/CommonInterfaces.ts (hunk 1 above)
Comment (kolya182) [reply src/common/CommonInterfaces.ts]: Initial `gas` type was defined wrong. It does not return value, it used as an interface.

File: src/common/TransactionParser.ts (hunk 2 above)
Comment (vikmeup) [reply src/common/TransactionParser.ts]: let's do it now, that's the whole point of this change

**Pull request merged**